khmer/_khmer.cc

```diff
@@ -238,6 +238,51 @@ typedef struct {
 
 
 static
+PyObject*
+khmer_Read_new(PyTypeObject * type, PyObject * args, PyObject * kwds)
+{
+    khmer_Read_Object * self;
+    self = (khmer_Read_Object *)type->tp_alloc(type, 0);
+    if (self != NULL) {
+        self->read = new Read;
+    }
+    return (PyObject *)self;
+}
+
+static
+int
+khmer_Read_init(khmer_Read_Object *self, PyObject *args, PyObject *kwds)
+{
+    const char * name{};
+    const char * annotations{};
+    const char * sequence{};
+    const char * quality{};
+    char *kwlist[5] = {
+        const_cast<char *>("name"), const_cast<char *>("sequence"),
+        const_cast<char *>("quality"), const_cast<char *>("annotations"), NULL
+    };
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|zzzz", kwlist,
```

Comment:
Agreed. Now we have `khmer.Read(name, seq)` vs `screed.Record(dict(name=name, sequence=sequence))`. Actually you can pass any kind of `dict` to `Record`.

Should we move `Read` to this "pass a dict as only arg" interface? I prefer the named parameters, but if you frequently turn dicts into `Read`s it will get tedious typing `Read(*my_dict)`.